src/primitives.jl

```diff
@@ -210,8 +210,18 @@ dropnull!(X::NullableVector) = deleteat!(X, find(X.isnull)).values # -> Vector
 
 Returns whether or not any entries of `X` are null.
 """
-anynull(X::Any) = any(_isnull, X)           # -> Bool
-anynull(X::NullableArray) = any(X.isnull)   # -> Bool
+anynull(X::Any) = any(_isnull, X)                              # -> Bool
+anynull(X::NullableArray) = any(X.isnull)                      # -> Bool
+function anynull{T}(X::AbstractArray{T})                       # -> Bool
+    u = isa(T, Union)
+    if u && !any(f -> eval(:($T.$f)) <: Nullable || Nullable <: eval(:($T.$f)), fieldnames(T))
```

Comment:
Also, 0.5:
```julia
!any(S -> S <: Nullable || Nullable <: S, T.types)
```
0.6:
```julia
!any(S -> S <: Nullable || Nullable <: S, Base.uniontypes(T))
```
Unfortunately neither of those nicer ones work on both versions.